src/schedule/zephyr_dp_schedule.c

```diff
@@ -360,14 +361,13 @@ static void dp_thread_fn(void *p1, void *p2, void *p3)
 			}
 		}
 
-		if (task->state == SOF_TASK_STATE_COMPLETED ||
-		    task->state == SOF_TASK_STATE_CANCEL)
-			break; /* exit the while loop, terminate the thread */
+		/* if true exit the while loop, terminate the thread */
+		task_stop = task->state == SOF_TASK_STATE_COMPLETED ||
+			task->state == SOF_TASK_STATE_CANCEL;
 
 		scheduler_dp_unlock(lock_key);
-	}
+	} while (!task_stop);
```

Comment:
Do we change the stop check in relation to holding the lock ?

Reply:
@lgirdwood no. No change there. Before the check was done in line 363 under the lock. Now the condition is also evaluated while holding the lock.